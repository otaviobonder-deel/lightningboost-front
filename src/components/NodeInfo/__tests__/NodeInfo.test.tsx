import React from "react";
import { render } from "@testing-library/react";
import { NodeInfo } from "../index";

/*
Usually a server mock would run here to gather a mocked graph
response from the api. But since I'm getting the info from
a static JSON file, there's no sense in testing it
*/

it("should match the snapshot", () => {
  const info = {
    publicKey: "test123",
    alias: "test",
    color: "#fff",
    visible: true,
    links: [
      {
        channelId: "123",
        node1: "test123",
        node2: "test345",
        capacity: "10000",
        color: "#fff",
      },
    ],
  };
  const graph2ScreenCoords = () => ({ x: 0, y: 0, z: 0 });
  const component = render(
    <NodeInfo graph2ScreenCoords={graph2ScreenCoords} info={info} />
  );
  expect(component.container).toMatchSnapshot();
});

it("should render without crashing", () => {
  const info = {
    publicKey: "test123",
    alias: "test",
    color: "#fff",
    visible: true,
    links: [
      {
        channelId: "123",
        node1: "test123",
        node2: "test345",
        capacity: "10000",
        color: "#fff",
      },
    ],
  };
  const graph2ScreenCoords = () => ({ x: 0, y: 0, z: 0 });
  render(<NodeInfo graph2ScreenCoords={graph2ScreenCoords} info={info} />);
});
