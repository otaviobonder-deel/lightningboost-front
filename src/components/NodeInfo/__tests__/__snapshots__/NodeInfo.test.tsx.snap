// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-container-1 MuiPaper-elevation1 MuiPaper-rounded"
    style="top: 20px; left: 0px;"
  >
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      Pubkey: 
      test123
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      Alias: 
      test
    </p>
  </div>
</div>
`;
